lib/semigrp.gi

```diff
@@ -450,9 +459,18 @@ function(S)
     fi;
 
     # determine D classes and sort according to rank.
-    layer:= List([1..DegreeOfTransformationSemigroup(S)], x->[]);
+    n := DegreeOfTransformationSemigroup(S);
+
+    if n = 0 then
+      # special case for the full transformation monoid on one point
+      Print("Rank 0: ");
+      displayDClass(GreensDClasses(S)[1]);
+      return;
+    fi;
+
+    layer:= List([1 .. n], x->[]);
     for D in GreensDClasses(S) do
-        Add(layer[RankOfTransformation(Representative(D))], D);
+        Add(layer[RankOfTransformation(Representative(D), n)], D);
```

Comment:
Again: this line uses 4 spaces for indentation, a few lines earlier you use 2: unify this?